### PR TITLE
[camera_web] Migrate to `package:web`

### PR DESCRIPTION
**Pull request by Rexios80**
Fixes https://github.com/flutter/flutter/issues/139748

Notes:
- Some fields are no longer nullable in `package:web`. We need to decide if we want to make them nullable again.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by Rexios80** [comment]
I figured out how to run the tests compiled to WASM. Working on resolving the issues.

**Comment by Rexios80** [comment]
Alright I think all the tests pass when compiled to WASM now

**Comment by Rexios80** [comment]
The example app is having this issue: https://github.com/flutter/flutter/issues/146539

**Comment by ditman** [comment]
> I figured out how to run the tests compiled to WASM. Working on resolving the issues.

@Rexios80 Please describe? :)

**Comment by Rexios80** [comment]
@ditman its just `flutter drive -d chrome --wasm`

Probably requires master channel

**Comment by Rexios80** [comment]
This is just a reminder to myself to merge with main and update https://github.com/flutter/packages/blob/main/script/configs/exclude_all_packages_app_wasm.yaml to remove this plugin from the exclusions

(away from dev computer right now)

**Comment by ditman** [comment]
Will work on this next, will try to get the update with main done and review locally. Thanks @Rexios80!

**Comment by ditman** [comment]
Force-pushed the rebase, but I think I broke:

```
integration_test/zoom_level_capability_test.dart is being served at http://localhost:7357
```

It seems to be timing out :/

**Comment by Rexios80** [comment]
I'm not sure what the Flutter team's policy is for rebase vs merge commits but I much prefer merge commits for the future if that's okay with you. It appears that the rebase invalidated my commit signing and we also lost easy access to the old CI runs.

**Comment by Rexios80** [comment]
@ditman It runs fine for me individually

**Comment by Rexios80** [comment]
Okay the test hangs locally too... but the Chrome dev console says the tests pass

```console
 This app is linked to the debug service: ws://127.0.0.1:64966/3_JgbVVF6ak=/ws
dart_sdk.js:37972 00:00 +0: CameraWebException sets all properties
dart_sdk.js:37972 00:00 +1: CameraWebException toString includes all properties
dart_sdk.js:37972 00:00 +2: (tearDownAll)
dart_sdk.js:37972 00:00 +3: All tests passed!
:7357/$dwdsSseHandler?sseClientId=InjectedClient-25923bcd-467a-457b-98b1-ba47d869a2ac:1 
        
        
       Failed to load resource: net::ERR_CONNECTION_REFUSED

```

Is this an issue with the test driver?

**Comment by Rexios80** [comment]
Related to https://github.com/flutter/flutter/issues/151561#issuecomment-2226200260 maybe?

**Comment by Rexios80** [comment]
This is happing in a PR without these changes so it's not this PR's fault

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8742583604102720209/+/u/Run_package_tests/drive_examples/stdout

**Comment by ditman** [comment]
> I much prefer merge commits for the future if that's okay with you. 

Yep, apologies @Rexios80, I normally only rebase in my own branches or things that are "one commit away" from landing, didn't mean to mess the branch too badly!

About running the tests in --wasm, it's a nice to have, but shouldn't be a blocker. Maybe the test that I saw hanging happened because my flutter `master` wasn't new enough to get the `--canvaskit` fixes (?).

I'll come back to this PR ASAP!

**Comment by Rexios80** [comment]
@ditman The reason I think WASM integration tests should block this is because I'm 100% sure they are failing even if the canvaskit tests pass. I think https://github.com/flutter/flutter/issues/146539 is the biggest issue right now for compiling plugins to WASM that have platform views. I'm willing to bet every single plugin that deals with platform views has failing WASM integration tests because of that issue.

**Comment by Rexios80** [comment]
Actually it looks like a possible fix for that issue was recently merged. Let me test it.

**Comment by Rexios80** [comment]
Welp `video_player_web` needs updated to `package:web` version `1.0.0` to test that. I'm really confused how the example is even resolving dependencies right now. Isn't `0.x.x` to `1.x.x` supposed count as a major version increase? But somehow the example is resolving `web: 1.0.0` even though `video_player_web` depends on a `0.x.x` version?????? Which is causing compile time errors. Is this known behavior? If not that's a pretty bad bug in dependency resolution.

**Comment by Rexios80** [comment]
I have some good news! The fix for platform view rendering seems to work for `google_maps_flutter` at least. You can see it in action here: https://beta.madadmaps.com/

I'm still having issues with `pointer_interceptor` when compiled to WASM. Maybe WASM integration tests would help catch that issue?

**Comment by Rexios80** [comment]
> Welp `video_player_web` needs updated to `package:web` version `1.0.0` to test that. I'm really confused how the example is even resolving dependencies right now. Isn't `0.x.x` to `1.x.x` supposed count as a major version increase? But somehow the example is resolving `web: 1.0.0` even though `video_player_web` depends on a `0.x.x` version?????? Which is causing compile time errors. Is this known behavior? If not that's a pretty bad bug in dependency resolution.

It's actually resolving the version of `video_player_web` before `package:web` was added which is... I guess technically correct? But quite unfortunate.

**Comment by ditman** [comment]
> I'm still having issues with pointer_interceptor when compiled to WASM. Maybe WASM integration tests would help catch that issue? Tracking here: https://github.com/flutter/flutter/issues/152109

This is being currently looked at (by @eyebrowsoffire), IIRC this is also affecting dart devtools, so it's a high priority!

> It's actually resolving the version of video_player_web before package:web was added which is... I guess technically correct? But quite unfortunate.

Yikes, I didn't see this one coming, I guess there can be multiple plugins with the same issue with the way that we version. I'll see what's up with the package:web 1 stuff.

**Comment by ditman** [comment]
```
Because every version of cross_file from path depends on web ^0.5.0
and every version of camera_web from path depends on web ^1.0.0,
cross_file from path is incompatible with camera_web from path.
```

I'm going to try and create a migration from `web:^0.5.0` to `^1.0.0`, so this change lands more smoothly.

**Comment by Rexios80** [comment]
@ditman You'll probably have to migrate all packages to web 1.0.0 to fix that error. That's just the first issue dependency resolution ran into. There's an issue to track that migration here: https://github.com/flutter/flutter/issues/151981

**Comment by Rexios80** [comment]
Getting

```console
UnimplementedError: CrossFile is not available in your current platform.
```

when running integration tests compiled to WASM. Caused by `cross_file` not supporting `web: 1.0.0` yet.

Blocked on
- [ ] https://github.com/flutter/packages/pull/7202

**Comment by ditman** [comment]
Landing this:

* https://github.com/flutter/packages/pull/7202

Should unblock this PR, I think!

**Comment by kevmoo** [comment]
https://github.com/flutter/packages/pull/7202 landed!

**Comment by Rexios80** [comment]
@ditman Is there an issue to track cleaning up code related to `script/configs/exclude_all_packages_app_wasm.yaml` after this PR lands?

**Comment by ditman** [comment]
@Rexios80 just created this one:

* https://github.com/flutter/flutter/issues/152716

**Comment by ditman** [comment]
(I have some time, I'm going to rebase this branch to make it match the rest of the repo!)

**Comment by ditman** [comment]
```
type 'Null' is not a subtype of type 'YamlList' in type cast
...
#6      PackageCommand._expandYamlInPackageList (package:flutter_plugin_tools/src/common/package_command.dart:292:8)
#7      PackageCommand.getExcludedPackageNames (package:flutter_plugin_tools/src/common/package_command.dart:298:9)
...
```

Hehe, looks like the tool doesn't like the yaml file to be completely empty!

**Comment by ditman** [comment]
<details>

<summary><tt>dart run flutter_plugin_tools.dart drive-examples --current-package --web --run-chromedriver --wasm</tt></summary>

```
============================================================
|| Running for packages/camera/camera_web
============================================================

Starting chromedriver on port 4444
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/zoom_level_capability_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/zoom_level_capability_test.dart on Web Server in debug mode...
Compiling integration_test/zoom_level_capability_test.dart for the Web...        24.5s
✓ Built build/web
integration_test/zoom_level_capability_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_service_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_service_test.dart on Web Server in debug mode...
Compiling integration_test/camera_service_test.dart for the Web...        24.5s
✓ Built build/web
integration_test/camera_service_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_web_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_web_test.dart on Web Server in debug mode...
Compiling integration_test/camera_web_test.dart for the Web...        28.1s
✓ Built build/web
integration_test/camera_web_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_metadata_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_metadata_test.dart on Web Server in debug mode...
Compiling integration_test/camera_metadata_test.dart for the Web...        23.6s
✓ Built build/web
integration_test/camera_metadata_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_web_exception_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_web_exception_test.dart on Web Server in debug mode...
Compiling integration_test/camera_web_exception_test.dart for the Web...        23.6s
✓ Built build/web
integration_test/camera_web_exception_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_test.dart on Web Server in debug mode...
Compiling integration_test/camera_test.dart for the Web...         25.6s
✓ Built build/web
integration_test/camera_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_options_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_options_test.dart on Web Server in debug mode...
Compiling integration_test/camera_options_test.dart for the Web...        24.0s
✓ Built build/web
integration_test/camera_options_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_bitrate_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_bitrate_test.dart on Web Server in debug mode...
Compiling integration_test/camera_bitrate_test.dart for the Web...        24.4s
✓ Built build/web
integration_test/camera_bitrate_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Running command: "flutter drive -d web-server --web-port=7357 --browser-name=chrome --wasm --driver test_driver/integration_test.dart --target integration_test/camera_error_code_test.dart" in /work/flutter/packages/packages/camera/camera_web/example
Resolving dependencies... 
Downloading packages... 
  _fe_analyzer_shared 72.0.0 (73.0.0 available)
  analyzer 6.7.0 (6.8.0 available)
  collection 1.18.0 (1.19.0 available)
  http_parser 4.0.2 (4.1.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  mocktail 0.3.0 (1.0.4 available)
  shelf 1.4.1 (1.4.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching integration_test/camera_error_code_test.dart on Web Server in debug mode...
Compiling integration_test/camera_error_code_test.dart for the Web...        24.2s
✓ Built build/web
integration_test/camera_error_code_test.dart is being served at http://localhost:7357
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge
devices for an improved development workflow.
All tests passed.
Application finished.
Stopping chromedriver
```

</details>

```console
------------------------------------------------------------
Run overview:
  packages/camera/camera_web - ran

Ran for 1 package(s)

No issues found!
```

**Comment by Rexios80** [comment]
@ditman thats great! Did you check the chrome console though? That was the only way I found the COEP error for the maps tests.

**Comment by ditman** [comment]
@Rexios80 I didn't, but I wouldn't expect COOP/COEP issues in this case, because we're not loading any data (images/scripts...) from an external site, as far as I know? (I can't get the --keep-browser-open thing to work in my linux box :P)

**Comment by Rexios80** [comment]
@ditman I wasn't expecting a COEP error that was just an example of something that didn't get caught by the CLI tooling

**Comment by ditman** [comment]
> @ditman I wasn't expecting a COEP error that was just an example of something that didn't get caught by the CLI tooling

@Rexios80 unfortunately, this can happen at any time, and we might not see those errors unless the error is blocking something that the test is actually looking at (imagine that the test blocks the loading of an image that you're expecting).

I wish the `flutter drive` output for web included the JS console contents at least :P

**Comment by Rexios80** [comment]
> I wish the flutter drive output for web included the JS console contents at least :P

Is this possible? One thing we really need to start tracking is all the issues with WASM test reporting.

**Comment by Rexios80** [comment]
@ditman I ran all the tests and there were no secret errors in the Chrome console

**Comment by ditman** [comment]
> > I wish the flutter drive output for web included the JS console contents at least :P
> 
> Is this possible? One thing we really need to start tracking is all the issues with WASM test reporting.

I think `flutter test` is able to extract the console logs and errors in web tests, I'd hope that to be possible in `flutter drive` as well, yes, but I haven't looked into how the tool is wired up too much 😅

**Comment by kevmoo** [comment]
`auto-submit`?

**Comment by Rexios80** [comment]
@ditman do you have any comments? I'm not sure you ever commented on the code.

**Comment by kevmoo** [comment]
He commented w/ his commits. 😀

**Review by ditman — COMMENTED** [review]
I don't see anything ultra-blocking in the package, but in general, this is the package that seems to have accrued more technical debt in this migration than any other, hence the extra comments/questions.

This package is in need of a good cleanup, I'm sure that there's a bunch more things that could be simplified that the ones that I pointed out in comments!

(I'll give it another go on Monday, see if I can turn any of my comments in actual changes before landing this!)

File: packages/camera/camera_web/lib/src/shims/dart_js_util.dart
Comment (Rexios80) [reply packages/camera/camera_web/lib/src/shims/dart_js_util.dart:6]: I'm pretty sure this file exists solely for mocking in tests

File: packages/camera/camera_web/lib/src/types/camera_error_code.dart
Comment (Rexios80) [reply packages/camera/camera_web/lib/src/types/camera_error_code.dart:82]: I think there was an issue switching on JS objects

Comment (kevmoo) [reply packages/camera/camera_web/lib/src/types/camera_error_code.dart:82]: https://pub.dev/documentation/web/latest/web/MediaError/MEDIA_ERR_ABORTED.html

These are (sadly) not const!

Comment (ditman) [reply packages/camera/camera_web/lib/src/types/camera_error_code.dart:82]: Ooooh shoot, I thought those were an enum!

Comment (kevmoo) [reply packages/camera/camera_web/lib/src/types/camera_error_code.dart:82]: Interop is annoying here

Comment (kevmoo) [reply packages/camera/camera_web/lib/src/types/camera_error_code.dart:82]: https://html.spec.whatwg.org/multipage/media.html#error-codes

Seems like the kind of thing we could either hard-wire from the IDL or...uh..."coerce" to be const

File: packages/camera/camera_web/lib/src/shims/dart_js_util.dart
Comment (ditman) [reply packages/camera/camera_web/lib/src/shims/dart_js_util.dart:6]: It's used for the sillies reason in `camera_service.dart`, and then in tests to inject the definitive hacks :P

File: packages/camera/camera_web/example/integration_test/camera_error_code_test.dart
Comment (ditman) [reply packages/camera/camera_web/example/integration_test/camera_error_code_test.dart:169]: Issue to remove `mocktail` is here:

https://github.com/flutter/flutter/issues/130757

File: packages/camera/camera_web/example/integration_test/helpers/mocks.dart
Comment (ditman) [reply packages/camera/camera_web/example/integration_test/helpers/mocks.dart:19]: This will be removed when we leave 0.5.x behind, as mentioned in the comment right above it. DUH!

Comment (kevmoo) [reply packages/camera/camera_web/example/integration_test/helpers/mocks.dart:19]: Well, it's 1.0.0 now...but...yeah

Comment (Rexios80) [reply packages/camera/camera_web/example/integration_test/helpers/mocks.dart:19]: @kevmoo We are supporting the range `0.5.0` through `2.0.0` for now to alleviate migration pains to web version `1.0.0`

Comment (kevmoo) [reply packages/camera/camera_web/example/integration_test/helpers/mocks.dart:19]: of course! sorry. I should have been more clear. Just the comment about `0.6.0` – we never had such a release. Carry on!

**Review by ditman — APPROVED** [review]
LGTM! I think this is good to go. Some pending changes have an associated issue so we don't forget to tackle them later.

**Comment by kevmoo** [comment]
`autosubmit`?

**Review by kevmoo — APPROVED** [review]
@ditman changes LGTM!

File: packages/camera/camera_web/example/integration_test/camera_service_test.dart
Comment (Rexios80) [inline packages/camera/camera_web/example/integration_test/camera_service_test.dart:93]: nit: I much prefer using trailing commas such as

```dart
expect(
  capturedConstraints?.video.dartify(),
  equals(options.video.toMediaStreamConstraints().dartify()),
);
expect(
  capturedConstraints?.audio.dartify(),
  equals(options.audio.toMediaStreamConstraints().dartify()),
);
```

But this is definitely non-blocking

Comment (ditman) [reply packages/camera/camera_web/example/integration_test/camera_service_test.dart]: Yep, good change! I forgot to reformat those, thanks for picking it up! Let's land this!

**Comment by ditman** [comment]
The failed test was a flake, I'll keep an eye on this to ensure it lands by EOD today.

**Comment by Rexios80** [comment]
@ditman is there an issue to track the Mac_arm64 flakes?

**Comment by ditman** [comment]
> @ditman is there an issue to track the Mac_arm64 flakes?

@Rexios80 can't remember off the top of my head, but mitigating them is definitely very high in the list of priorities of the team.

**Pull request merged**
